### PR TITLE
Added a basic search endpoint, started parsing.

### PR DESCRIPTION
**Pull request by doctor3w**
Added the Draft pod for endpoints, started parsing code by adding FeedKit. I think our best move is to make our own parsing, this way we can parse directly into our own models.

### CODE REVIEW

File: Recast/Models.swift [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    
+    private func parse(items: [RSSFeedItem]?) -> [Episode] {
+        guard let items = items else { return [] }
+        // TODO: Implement this
+        return []
+    }
+    
+    func parse(shouldParseItems: Bool) {
+        let parser = FeedParser(URL: partial.feedUrl)
+        parser.parseAsync(queue: DispatchQueue.global(qos: .userInitiated)) { (result) in
```
Comment (mindylou) [inline Recast/Models.swift]: remove parentheses

File: Recast/SearchExampleViewController.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        SearchEndpoint(parameters: ["term": "#DORK", "media": "podcast", "limit": 1]).run()
+            .success { (response: SearchResults) in
+                guard response.resultCount > 0,
+                    let partial = response.results.first else { return }
+                Podcast.loadFull(from: partial, success: { podcast in
+                    print(podcast.title!)
+                    print(podcast.description!)
+                    print(podcast.link!)
+                    print(podcast.collectionName ?? "")
+                    print(podcast.items[1].title ?? "")
```
Comment (mindylou) [inline Recast/SearchExampleViewController.swift]: is there a safer way to get stuff out of `items` aside from indexing into the array? is it guaranteed to have the 1st index?

Comment (mindylou) [reply Recast/SearchExampleViewController.swift]: also, remove these print statements when you merge in

File: Recast/iTunesPodcasts/Models/RSS/mapAttributes.swift [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    ///   - path: The path of feed's element.
+    func map(_ attributes: [String : String], for path: RSSPath) {
+
+        switch path {
+            
+        case .rssChannelItem:
+
+            self.items.append(Episode())
+
+//        case .rssChannelImage:
```
Comment (mindylou) [inline Recast/iTunesPodcasts/Models/RSS/mapAttributes.swift]: remove comments if these fields are unnecessary

File: Recast/iTunesPodcasts/Models/RSS/mapCharacters.swift [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        case .rssChannelCategory:                                   self.categories?
+            .append(string)
+        case .rssChannelGenerator:                                  self.generator                                                  = self.generator?.appending(string) ?? string
+        case .rssChannelDocs:                                       self.docs                                                       = self.docs?.appending(string) ?? string
+        case .rssChannelRating:                                     self.rating                                                     = self.rating?.appending(string) ?? string
+        case .rssChannelTTL:                                        self.ttl                                                        = Int(string)
+        case .rssChannelImageURL:                                   self.image                                                 = URL(string: string)
+//        case .rssChannelImageURL:                                   self.image?.url                                                 = self.image?.url?.appending(string) ?? string
```
Comment (mindylou) [inline Recast/iTunesPodcasts/Models/RSS/mapCharacters.swift]: just a question... what's the difference between `image` and `image.url` in this case?

File: Recast/iTunesPodcasts/Models/iTunes/iTunesNamespace.swift [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+    /// Use the <itunes:episode> tag in conjunction with the <itunes:season> tag
+    /// to indicate the order an episode should be presented within a season.
+    public var episode: Int?
+}
+
+// MARK: - Equatable
+//
+//extension ITunesNamespace: Equatable {
+//
+//    public static func ==(lhs: ITunesNamespace, rhs: ITunesNamespace) -> Bool {
```
Comment (mindylou) [inline Recast/iTunesPodcasts/Models/iTunes/iTunesNamespace.swift]: delete this if we don't need it

**Review by mindylou — CHANGES_REQUESTED** [review]
looks good! just remove some of the commented out code if we don't need it to reduce clutter then i'll approve

File: Recast/iTunesPodcasts/Models/RSS/mapCharacters.swift (hunk 4 above)
Comment (doctor3w) [reply Recast/iTunesPodcasts/Models/RSS/mapCharacters.swift]: Image in RSS can contain more subfields than just a URL, mainly for not being able to load the image (alt text for HTML). We don't really need that.

File: Recast/iTunesPodcasts/Models/iTunes/iTunesNamespace.swift (hunk 5 above)
Comment (doctor3w) [reply Recast/iTunesPodcasts/Models/iTunes/iTunesNamespace.swift]: uncommented, we actually want that.

File: Recast/SearchExampleViewController.swift (hunk 2 above)
Comment (doctor3w) [reply Recast/SearchExampleViewController.swift]: Oh this file is just an example so people can see how to use it. Those '!' are actually unnecessary and the items is all optional, so we will want our views to be able to handle these cases

**Comment by doctor3w** [comment]
Made changes.

**Review by mindylou — DISMISSED** [review]

File: Recast/iTunesPodcasts/Models/RSS/Enclosure.swift [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+            let type = attributes["type"],
+            let url = URL(string: attributes["url"] ?? ""),
+            let length = Int(attributes["length"] ?? "") else {
+                return nil
+        }
+        
+        let audio_types = ["audio/x-m4a", "audio/mpeg"]
+        let video_types = ["video/quicktime", "video/mp4", "video/x-m4v"]
+        let pdf_types = ["application/pdf"]
+        if audio_types.contains(type)  {
```
Comment (jackthmp) [inline Recast/iTunesPodcasts/Models/RSS/Enclosure.swift]: is it better to replace this with a switch? ex:
```
switch type:
  case _ where audio_types.contains(type):
```

**Review by jackthmp — DISMISSED** [review]
LGTM, just a small style change

File: Recast/iTunesPodcasts/Models/RSS/Enclosure.swift [hunk 7]
```diff
@@ -13,31 +13,37 @@ public enum Enclosure {
...
+        switch type {
+        case .m4a, .mpeg:
+            self = .audio(url: url, length: length, type: type)
+        case .quicktime, .mp4, .m4v:
+            self = .video(url: url, length: length, type: type)
+        case .pdf:
             self = .pdf(url: url, length: length)
-            return
         }
         return nil
```
Comment (jackthmp) [inline Recast/iTunesPodcasts/Models/RSS/Enclosure.swift]: this `return nil` will always be executed, you sure this is what we want from our init? or should that be in `case default:`

**Review by jackthmp — CHANGES_REQUESTED** [review]

File: Recast/iTunesPodcasts/Models/RSS/Enclosure.swift (hunk 7 above)
Comment (doctor3w) [reply Recast/iTunesPodcasts/Models/RSS/Enclosure.swift]: ahh nah I need to insert regular returns

**Review by jackthmp — DISMISSED** [review]

**Review by mindylou — DISMISSED** [review]

**Review by mindylou — APPROVED** [review]

**Pull request merged**